libmogwai-schedule/tests/scheduler.c

```diff
@@ -386,6 +394,391 @@ test_scheduler_properties (Fixture       *fixture,
   g_assert_nonnull (peer_manager);
 }
 
+/* Convenience method to create a new schedule entry and set its priority. */
+static MwsScheduleEntry *
+schedule_entry_new_with_priority (const gchar *owner,
+                                  guint32      priority)
+{
+  g_autoptr(MwsScheduleEntry) entry = mws_schedule_entry_new (owner);
+  mws_schedule_entry_set_priority (entry, priority);
+  return g_steal_pointer (&entry);
+}
+
+/* Assert that when all the entries in @expected_scheduling_order are added to
+ * the scheduler, they are scheduled in the given order. This is checked by
+ * adding them all, checking which one entry is active, removing it, then
+ * checking which entry is made active next, etc. This checks all signal
+ * emissions. It requires the scheduler to be empty beforehand, and to have a
+ * max-active-entries limit of 1. */
+static void
+assert_scheduling_order (Fixture                 *fixture,
+                         const MwsScheduleEntry **expected_scheduling_order,
+                         gsize                    n_entries)
+{
+  g_autoptr(GError) local_error = NULL;
+
+  g_assert (n_entries > 0);
+  g_assert (g_hash_table_size (mws_scheduler_get_entries (fixture->scheduler)) == 0);
+  guint max_active_entries;
+  g_object_get (fixture->scheduler, "max-active-entries", &max_active_entries, NULL);
+  g_assert (max_active_entries == 1);
```

Comment:
If this is used for the tests only you can use the `g_assert_cmp*` functions above, right?

Reply:
I deliberately used `g_assert()` here (and just above) because it’s asserting that the setup of the fixture is correct — it’s not actually an assertion which is part of the test.